### PR TITLE
Extract permitted params constant in api controllers

### PR DESCRIPTION
**Pull request by mjankowski**
Follow-up on https://github.com/mastodon/mastodon/pull/30652

This is a batch of all the "simple" ones, where it's just a 1:1 extraction of the keys into a constant, and the addition of a slice.

This probably slightly more than half of the remaining ones from WIP branch, but the rest are marginally more complex. Will continue to rebase and work through them.

### CODE REVIEW

**Comment by ClearlyClaire** [comment]
I've been accepting the previous PRs, but if the primary goal is to avoid tests failing… maybe we should actually fix the tests rather than use `slice`?

I actually kind of prefer having the parameters defined directly where they are used (though that preference quickly becomes irrelevant when the parameters list is duplicated or very lengthy).

**Comment by mjankowski** [comment]
> I've been accepting the previous PRs, but if the primary goal is to avoid tests failing… maybe we should actually fix the tests rather than use `slice`?

Yes, when we enable https://github.com/mastodon/mastodon/pull/30377 we get a bunch of failures. To look at one of the simpler examples, in `api/v1/admin/accounts` the request spec will fail with:

```
  1) Accounts GET /api/v1/admin/accounts with limit param returns only the requested number of accounts
     Failure/Error: params.permit(*FILTER_PARAMS)
     
     ActionController::UnpermittedParameters:
       found unpermitted parameter: :limit
     # ./app/controllers/api/v1/admin/accounts_controller.rb:111:in `filter_params'
     # ./app/controllers/api/v1/admin/accounts_controller.rb:115:in `translated_filter_params'
     # ./app/controllers/api/v1/admin/accounts_controller.rb:107:in `filtered_accounts'
     # ./app/controllers/api/v1/admin/accounts_controller.rb:99:in `set_accounts'
```

In that particular case, all the params are just top-level (not nested) params, so the "filter params" (already in a constant there) and the "pagination params" (just `limit`) are all side by side by with each other, so the `permit` call (to allow through the filter-specific ones from the constant) still has `limit` in it, which is why the `slice` there is useful.

All that said -- I have just now for the first time noticed (while reviewing this code to write that explanation) that we have a `params_slice` method in the api base controller which is for this exact purpose, but which is used inconsistently ... I bet that just apply that to more areas would solve a lot of this.

Let me either rework this one or submit another PR with that approach, and we'll see how many are left after that.

> I actually kind of prefer having the parameters defined directly where they are used (though that preference quickly becomes irrelevant when the parameters list is duplicated or very lengthy).

Basically agree. I'd say:

- Small number of params, just do the inline `params.require(...).permit(...)` style
- Once it gets large, pull out constant
- Even if it's small/medium, if you need to repeat the params in a `slice(...).permit(...)` style, maybe also pull out constant
- If you have 100 controllers and 98 of them have constants and 2 dont ... just pull out those last two for consistency

Anyway ... stand by for update here one way or another, I'll see what I can find with the existing `params_slice` method that I had missed earlier.
